### PR TITLE
(feat) re-use upcoming appointment work to mark appointments as `Checked In`

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary
re-use upcoming appointment work to mark appointments as `Checked In`

### CODE REVIEW

File: packages/esm-patient-chart-app/translations/en.json [hunk 1]
```diff
@@ -5,9 +5,9 @@
   "all": "All",
   "allEncounters": "All encounters",
   "Allergies dashboard": "Allergies dashboard",
-  "appointmentEdited": "Appointment edited",
+  "appointmentCheckedIn": "Appointment checked",
```
Comment (mseaton) [inline packages/esm-patient-chart-app/translations/en.json:8]: checked in?

File: packages/esm-patient-chart-app/src/visit/hooks/useUpcomingAppointments.tsx [hunk 2]
```diff
@@ -17,13 +18,14 @@ export interface AppointmentPayload {
   dateHonored?: string;
 }
 
-export function saveAppointment(appointment: AppointmentPayload, abortController: AbortController) {
-  return openmrsFetch(`${restBaseUrl}/appointment`, {
+export const updateAppointmentStatus = async (toStatus: string, appointmentUuid: string) => {
+  const abortController = new AbortController();
```
Comment (mseaton) [inline packages/esm-patient-chart-app/src/visit/hooks/useUpcomingAppointments.tsx:22]: Previously this took in an abortController, now you are instantiating one.  I think there is a `useAbortController` hook in esm-framework.  Perhaps you are meant to use that? @ibacher / @denniskigen ?

File: packages/esm-patient-chart-app/src/visit/hooks/useUpcomingAppointments.tsx [hunk 3]
```diff
@@ -17,13 +18,14 @@ export interface AppointmentPayload {
...
+  const statusChangeTime = dayjs(new Date()).format(omrsDateFormat);
```
Comment (mseaton) [inline packages/esm-patient-chart-app/src/visit/hooks/useUpcomingAppointments.tsx:23]: Do you really need to instanatiate and format the date?  Can you not just submit a `new Date()` in the body?

File: packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx [hunk 4]
```diff
@@ -375,28 +375,16 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
...
-                      }
+                  updateAppointmentStatus('CheckedIn', upcomingAppointment?.uuid).then(
+                    () => {
+                      mutateCurrentVisit();
+                      mutateVisits();
+                      showSnackbar({
+                        isLowContrast: true,
+                        kind: 'success',
+                        subtitle: t('appointmentMarkedChecked', 'Appointment marked as Checked In'),
+                        title: t('appointmentCheckedIn', 'Appointment checked'),
```
Comment (mseaton) [inline packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx:386]: checked in?

**Review by mseaton — COMMENTED** [review]
I don't really know the background of this ticket, defer to @mogoodrich on that, but made a few comments within the PR.

File: packages/esm-patient-chart-app/src/visit/hooks/useUpcomingAppointments.tsx (hunk 3 above)
Comment (mogoodrich) [reply packages/esm-patient-chart-app/src/visit/hooks/useUpcomingAppointments.tsx:23]: I do think you need to format the date into the OMRS date format, but you probably don't need to do the new Date(), ie you could just do dayjs().format(omrsDateFormat), but i might be wrong about that.

**Review by mogoodrich — APPROVED** [review]
Generally looks good to me, though note @mseaton 's questions on the abort controller and comments about the "checked" text.

**Pull request merged**